libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -1063,65 +866,66 @@ void AC_AutoTune::load_intra_test_gains()
     attitude_control->bf_feedforward(true);
     if (roll_enabled()) {
         attitude_control->get_rate_roll_pid().kP(orig_roll_rp);
-        attitude_control->get_rate_roll_pid().kI(orig_roll_rp*AUTOTUNE_PI_RATIO_FOR_TESTING);
+        attitude_control->get_rate_roll_pid().kI(get_intra_test_ri(axis));
         attitude_control->get_rate_roll_pid().kD(orig_roll_rd);
         attitude_control->get_rate_roll_pid().ff(orig_roll_rff);
         attitude_control->get_rate_roll_pid().filt_T_hz(orig_roll_fltt);
         attitude_control->get_rate_roll_pid().slew_limit(orig_roll_smax);
         attitude_control->get_angle_roll_p().kP(orig_roll_sp);
+        attitude_control->set_accel_roll_max_cdss(orig_roll_accel);
```

Comment:
So we have as slight change in behaviour but I think it is an improvement.
AC_AutoTune::load_intra_test_gains() is setting the max acceleration to the original acceleration while in master it would be zero (meaning no max acceleration).

I can imagine that during the return to level the response could theoretically be a little slower.

FYI @lthall 

FYI @lthall

Reply:
moved backup and load gains to subclasses